src/geometry/rectangle.rs

```diff
@@ -0,0 +1,186 @@
+    /// ```*/
+    fn to_global(&self, input_point: P3)-> P3{
+        self.to_global * input_point
+    }
+    
+    /*
+    /// Converts a point in the local refernce frame of the sensor to the global reference frame.
+    /// 
+    /// # Examples
+    /// 
+    /// ```
+    /// use nalgebra as na;
+    /// use na::Point3;
+    /// use kalman_rs::geometry::rectangle::Rectangle;
+    /// use kalman_rs::sensor_traits::Transform;
+    /// 
+    /// let base = 3.0;
+    /// let height = 3.0;
+    /// let tfm_matrix : na::Matrix4<f64>= na::Matrix4::new(1.0,0.0,0.0,0.0,  0.0,1.0,0.0,0.0,  0.0,0.0,1.0,0.0, 0.0,0.0,0.0,1.0);
+    /// let mut rectangle_sensor = Rectangle::new(base, height, tfm_matrix).unwrap();
+    /// 
+    /// let global_point = rectangle_sensor.to_local(na::Point3::new(6.0, 3.0, 5.0));
+    /// ```*/
+    fn to_local(&self, input_point: P3) -> P2{
+        let local = self.to_local * input_point;
+        return P2::new(local.x, local.y)
+    }
+
+    /*
+    /// Checks if a local point is contained within the bounds of a sensor.
+    /// 
+    /// # Examples
+    /// ```
+    /// use nalgebra as na;
+    /// use na::Point3;
+    /// use kalman_rs::sensor_traits::Transform;
+    /// use kalman_rs::geometry::rectangle::Rectangle;
+    /// 
+    /// let base = 3.0;
+    /// let height = 3.0;
+    /// let tfm_matrix : na::Matrix4<f64>= na::Matrix4::new(1.0,0.0,0.0,0.0,  0.0,1.0,0.0,0.0,  0.0,0.0,1.0,0.0, 0.0,0.0,0.0,1.0);
+    /// let mut rectangle_sensor = Rectangle::new(base, height, tfm_matrix).unwrap();
+    /// 
+    /// let is_point_on_sensor = rectangle_sensor.contains_from_local(&na::Point2::new(1.0, 6.0));
+    /// ```*/
+    fn contains_from_local(&self, input: &P2) -> bool {
+        
+        if (input.x.abs() < self.half_base.abs()) && (input.y.abs() < self.half_height.abs()) {
+            true
+        }
+        else {
+            false
+        }
+    }
+}
+
+
+impl Plane for Rectangle{
+
+    /*
+    /// Check if a given point is located on the same plane as the sensor
+    /// NOTE: `plane()` must be called becuase the normal vector is not currently known
+    /// # Examples
+    /// 
+    /// ```
+    /// use nalgebra as na;
+    /// use na::Point3;
+    /// use kalman_rs::sensor_traits::Plane;
+    /// use kalman_rs::geometry::rectangle::Rectangle;
+    /// 
+    /// let base = 3.0;
+    /// let height = 3.0;
+    /// let tfm_matrix : na::Matrix4<f64>= na::Matrix4::new(1.0,0.0,0.0,0.0,  0.0,1.0,0.0,0.0,  0.0,0.0,1.0,0.0, 0.0,0.0,0.0,1.0);
+    /// let mut rectangle_sensor = Rectangle::new(base, height, tfm_matrix).unwrap();
+    /// 
+    /// let on_plane = rectangle_sensor.on_plane(&na::Point3::new(1.0, 3.0, 0.0)); //true
+    /// ```*/
+    fn on_plane(&self, input_point: &P3) -> bool {
+        let pv : Vec3= P3::new(0.0, 0.0, 0.0) - input_point;
+        //TODO : this function should probably not return result
+        if self.normal.dot(&pv) == 0.0 {
```

Comment:
Exactly zero is probably problematic, because it's a floating point operation. You should check if it is below a certain small epsilon > 0.

Reply:
Added an epsilon of 0.0005